src/components/Map.tsx

```diff
@@ -66,22 +73,44 @@ const Map = ({
   interactive = true,
   onFeatureSelected = () => {},
   showNoLocation,
+  showNearbyHotspots = false,
   ...props
 }: Props) => {
+  const dispatch = useAppDispatch()
   const colors = useColors()
   const { t } = useTranslation()
   const map = useRef<MapboxGL.MapView>(null)
   const camera = useRef<MapboxGL.Camera>(null)
   const [loaded, setLoaded] = useState(false)
   const [userCoords, setUserCoords] = useState({ latitude: 0, longitude: 0 })
   const [centerOffset, setCenterOffset] = useState(0)
+  const styles = useMemo(() => makeStyles(colors), [colors])
+
+  const {
+    networkHotspots: { networkHotspots },
+  } = useSelector((state: RootState) => state)
+
+  useAsync(async () => {
+    if (showNearbyHotspots && loaded) {
+      const bounds = await map.current?.getVisibleBounds()
+      if (bounds) {
+        dispatch(fetchNetworkHotspots(bounds))
+      }
+    }
+  }, [loaded, dispatch])
 
   const onRegionDidChange = useCallback(async () => {
     if (onMapMoved) {
       const center = await map.current?.getCenter()
       onMapMoved(center)
     }
-  }, [onMapMoved])
+    if (showNearbyHotspots) {
```

Comment:
So does this get called whenever you move the map?

Reply:
yeah it'll load whatever is in the visible map bounds which can be quite a few if you zoom out. it seemed to perform ok for me still though. and once it has them it keeps them in temporary memory